### PR TITLE
Pagination first cut

### PR DESCRIPTION
**Pull request by pyotrritchie**
Required upgrade of common facade to support a paginated request to the db hence the upgrade from 3 to 4.

Material ui was upgraded from 3.4 to 3.9 which I required to fix a bug in the drop down select used by the pagination table actions component.

Created a resource object to represent a paginated resource. Since the out of the box de/serialization library slipped up with the original IPageList object... 

PageList.Core is the library I chose to manage the actual pagination from what's returned by EF when we fetch all objects.. It's fairly lightweight and can be easily changed if we choose to do so.. It would probably be just as easy to do it ourselves too.. for example using the functions Skip(noOfRowsToDisplay * pageNumber).Take(noOfRowsToDisplay) and then gathering info like totalNoOfItems.. 

SalesPackagePaginatedResource could inherit from an abstract paginated resource.. maybe do that next time we do pagination for another view.

Worth noting that the pagination can work pointing at an end point that returns all items, need to put a slice function that cuts the array at the appropriate points given no of rows and page number.

### CODE REVIEW

File: src/Service.Host/.eslintrc.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
         "react/jsx-filename-extension": 0,
         "spaced-comment": "off",
         "react-hooks/rules-of-hooks": "error",
-        "react-hooks/exhaustive-deps": "error"
+        "react-hooks/exhaustive-deps": "error",
+        "react/destructuring-assignment": ["always", { "ignoreClassFields": true }]
```
Comment (lewisrenfrew) [inline src/Service.Host/.eslintrc.json]: Not really sure what this is... Is there somewhere you thought this rule had to be violated? 
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/destructuring-assignment.md

File: src/Service.Host/client/src/configureStore.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ import reducer from './reducers';
 import authorization from './middleware/authorization';
 import itemCreated from './middleware/itemCreated';
 
-const composeEnhancers = window['__REDUX_DEVTOOLS_EXTENSION_COMPOSE__'] || compose;
+// eslint-disable-next-line no-underscore-dangle
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/configureStore.js]: remove?

File: src/Service.Host/.eslintrc.json (hunk 1 above)
Comment (pyotrritchie) [reply src/Service.Host/.eslintrc.json]: It should actually be removed, I was trying to get something working but it's no longer needed to run without warnings

File: src/Service.Host/client/src/configureStore.js (hunk 2 above)
Comment (pyotrritchie) [reply src/Service.Host/client/src/configureStore.js]: I've used dot notation to get rid of another eslinter error and then i got the one above.. it's one or the other

**Pull request merged**